src/hooks/SubgraphQuery.ts

```diff
@@ -0,0 +1,51 @@
+import {
+  EntityKey,
+  formatGraphQuery,
+  formatGraphResponse,
+  GraphQueryOpts,
+  SubgraphEntities,
+  SubgraphQueryReturnTypes,
+} from '../utils/graph'
+import { useQuery, UseQueryOptions } from 'react-query'
+import axios from 'axios'
+
+const subgraphUrl = process.env.REACT_APP_SUBGRAPH_URL
+
+export default function useSubgraphQuery<E extends EntityKey>(
+  opts: GraphQueryOpts<E>,
+  reactQueryOptions?: UseQueryOptions<
+    SubgraphEntities[E][],
+    unknown, // Specific error type?
+    SubgraphEntities[E][],
+    readonly [string, GraphQueryOpts<E>]
+  >,
+) {
+  if (!subgraphUrl) {
+    // This should _only_ happen in development
+    throw new Error('env.REACT_APP_SUBGRAPH_URL is missing')
+  }
+  return useQuery<
+    SubgraphEntities[E][],
+    unknown, // Specific error type?
+    SubgraphEntities[E][],
+    readonly [string, GraphQueryOpts<E>]
+  >(
+    ['subgraph-query', opts],
```

Comment:
would it make sense to be able to override `subgraph-query` tag here? not entirely sure what's happening with that first argument to `useQuery`

Reply:
@peripheralist Sure, I can explain.

The first argument to React Query is a key. This can either be a single string or an array of serializable values. These are how multiple `useQuery`s with the same key on the same page can share state. I've used this to my advantage countless times, mostly to create seemingly isolated components that end up sharing data, without having to create and setup (with the right `value` at the right time) new React Contexts.

The keys are diff'd deterministically (more on that [in their docs](https://react-query.tanstack.com/guides/query-keys)) I've actually never thought of allowing a customizable query key base and I don't think it's very useful. I tend to create query keys as: what's the organizational or business "type" of data, and what are the parameters that make _this_ one unique? So in the case of `useSubgraphQuery`, you're asking for a subgraph query; so the key base (index 0) is "subgraph-query", and the parameters that make _that_ instance unique is the query structure itself.

Reply:
got it, thanks for the explanation. sounds like we can keep it as you have it for now 👍